### PR TITLE
build: Add build and publish pipeline by Docker

### PR DESCRIPTION
**Pull request by nqngo**
Add github workflow to build and publish Docker image.

Switch to using python-3.12-alpine to reduce the image size and vulnerability factors.

Poetry install now requires a README.md for package, add that into the Docker build pipeline.

Use explicit module naming to make sure the app can run properly in production.

### CODE REVIEW

**Review by rodonguyen — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM python:3.12 as build
+FROM python:3.12-alpine as build
```
Comment (samhwang) [inline Dockerfile:1]: Just keep it as debian based. Alpine is way too hard to work with, given the size tradeoff. Chances are we will need to install build tools to build the project anyway.

File: Dockerfile [hunk 2]
```diff
@@ -10,13 +10,13 @@ ENV POETRY_NO_INTERACTION=1 \
 
 RUN pip install poetry==${POETRY_VERSION}
 
-COPY pyproject.toml poetry.lock ./
+COPY pyproject.toml poetry.lock README.md ./
```
Comment (samhwang) [inline Dockerfile:13]: Does the process need the readme file?

Comment (nqngo) [reply Dockerfile:13]: yes. It needs a README.md file to for `poetry install` to works now

File: README.md [hunk 3]
```diff
@@ -1,38 +1,14 @@
...
----
-
-## Discord Bot Development
-
-### Getting started
+### Setting up the bot in Discord
 
 - [Create a bot in discord](https://interactions-py.github.io/interactions.py/Guides/02%20Creating%20Your%20Bot/).
-- Add that bot to a test server.
+- Add that bot to your server server.
```
Comment (samhwang) [inline README.md:10]: typo

File: Dockerfile (hunk 1 above)
Comment (nqngo) [reply Dockerfile]: Switched back to Debian.

**Review by samhwang — APPROVED** [review]

**Pull request merged**